### PR TITLE
Http parser

### PR DESCRIPTION
**Pull request by jteo1**
parse http request into its method, url, fields, and body.

### CODE REVIEW

File: http_parser.cc [hunk 1]
```diff
@@ -0,0 +1,104 @@
+#include "http_parser.h"
+#include <string>
+#include <vector>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+HttpParser* HttpParser::MakeHttpParser(const char* const raw_req)
```
Comment (dstephan1) [inline http_parser.cc]: Is this safe? Are you guaranteed that a request ends with a \0?

Comment (jteo1) [inline http_parser.cc]: i think this should be dealt with in connection.cc when boost reads the request into the data_ member variable

Comment (kevinazhu) [inline http_parser.cc]: In connection.cc, data has a max length of 1024. So no matter what, it'll have a \0, but yes, we are limited to a 1024 byte request right now....

Comment (dstephan1) [inline http_parser.cc]: i mean are we guaranteed there isn't garbage between 50-1023

**Comment by dstephan1** [comment]
Maybe use tabs instead of 4 spaces.

File: http_parser.cc (hunk 1 above)
Comment (kevinazhu) [inline http_parser.cc]: Just tested with setting the max length to 10 instead, and it seems to still work fine

Comment (kevinazhu) [inline http_parser.cc]: I'm pretty sure async_read_some when it has the max size takes care of this for us.

File: http_parser.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class HttpParser
+{
+public:
+    //factory
+    static HttpParser* MakeHttpParser(const char* const raw_req);
+
+    //getters
+    std::string get_path();
+    std::string get_method();
+    std::unordered_map<std::string, std::string> get_fields();
```
Comment (dstephan1) [inline http_parser.h]: This is a little strange. Every get_fields() call copies an entire map. There will probably only be one call of this per http parse, but it still seems a little strange.

Comment (jteo1) [inline http_parser.h]: okay, I'll change it to dynamic allocation and add a destructor

**Pull request merged**